src/pkg/cluster/data.go

```
@@ -59,103 +56,110 @@ func (c *Cluster) HandleDataInjection(ctx context.Context, wg *sync.WaitGroup, d
-				continue iterator
+				// Do the actual data injection
+				if err := exec.CmdWithPrint(shell, append(shellArgs, cpPodCmd)...); err != nil {
+					return fmt.Errorf("could not coy data ino the pod %s: %w", pod.Name, err)
+				}
+
+				// Leave a marker in the target container for pods to track the sync action
+				cpPodCmd = fmt.Sprintf("%s -C %s %s | %s -- %s",
+					tarCmd,
+					componentPath.DataInjections,
+					config.GetDataInjectionMarker(),
+					kubectlCmd,
+					untarCmd,
+				)
+
+				if err := exec.CmdWithPrint(shell, append(shellArgs, cpPodCmd)...); err != nil {
+					return fmt.Errorf("could not save the Zarf sync completion file after injection into pod %s: %w", pod.Name, err)
+				}
 			}
-		}
 
-		// Do not look for a specific container after injection in case they are running an init container
-		podOnlyTarget := podLookup{
-			Namespace: data.Target.Namespace,
-			Selector:  data.Target.Selector,
-		}
+			// Do not look for a specific container after injection in case they are running an init container
+			podOnlyTarget := podLookup{
+				Namespace: data.Target.Namespace,
+				Selector:  data.Target.Selector,
+			}
 
-		// Block one final time to make sure at least one pod has come up and injected the data
-		// Using only the pod as the final selector because we don't know what the container name will be
-		// Still using the init container filter to make sure we have the right running pod
-		_ = waitForPodsAndContainers(ctx, c.Clientset, podOnlyTarget, podFilterByInitContainer)
+			// Block one final time to make sure at least one pod has come up and injected the data
+			// Using only the pod as the final selector because we don't know what the container name will be
+			// Still using the init container filter to make sure we have the right running pod
+			_, err = waitForPodsAndContainers(ctx, c.Clientset, podOnlyTarget, podFilterByInitContainer)
+			if err != nil {
+				return nil
```

Is there a reason we are returning `nil` here rather than the `err`?

---

I think my initial logic was that errors were not handled before so lets not do it now. That was stupid the more I think about it though.